### PR TITLE
odc-6192-updated the test-headless script in dev-console/package.json file

### PR DESCRIPTION
**Pull request by makambalaji**
**Description:**
To avoid the merging issue on ODC-6023, this piece of code changes is applying as part of this PR. This task, helps us to execute all add flow smoke test scenarios in CI

**Jira Id:**
https://issues.redhat.com/browse/ODC-6192

**Refactoring criteria for approving Automation PR:**

- [x] update the CI status in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands:**
Example:
```
    export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
    BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
    BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.8-042801.devcluster.openshift.com
    export BRIDGE_KUBEADMIN_PASSWORD
    export BRIDGE_BASE_ADDRESS
    oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
    oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
    oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
    ./test-cypress.sh -p dev-console -h true
```

**Screen shots**:
image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by gajanan-more — APPROVED** [review]
/lgtm

**Comment by jerolimov** [comment]
@makambalaji Let us see if this merges. I opened already #9528 after our problems with this change and raised an issue where I think this still block this change. See initial comment on that PR.

**Comment by jerolimov** [comment]
Tests doesn't pass.
/hold

**Comment by makambalaji** [comment]
/unhold
Issue got fixed : https://github.com/openshift/console/pull/9637

**Comment by makambalaji** [comment]
/bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1987458

File: frontend/packages/dev-console/integration-tests/cypress.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "screenshotsFolder": "../../../gui_test_screenshots/cypress/screenshots",
   "videosFolder": "../../../gui_test_screenshots/cypress/videos",
   "env": {
-    "TAGS": "@smoke and not (@manual or @to-do or @un-verified)",
+    "TAGS": "(@pre-condition or @smoke) and not (@manual or @to-do or @un-verified or @bug)",
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/cypress.json:22]: Don't we need regression tag?

Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/cypress.json:22]: Regression takes more time. First lets integrate this and later on we can add regression test scenarios as well based on the execution time

**Review by jerolimov — COMMENTED** [review]
@makambalaji I can accept the changes for the moment because this blocks #9519.

But I strongly recommend to check my comments above. Rename `@bug` so something unambiguously and check how we can improve and stabilize the topology conditions.

With the new user preferences the UI will be even more dynamic and could depend on the result of other tests (or an user which uses a cluster at the same moment).

I would recommend to check all topology gherkin sentences that they include also the graph or view informational:

```diff
-And user is on the topology page
+And user is on the topology page in graph mode
```

**Comment by jerolimov** [comment]
PR addresses integration tests only.

/label px-approved
/label docs-approved
/label qe-approved

**Comment by jerolimov** [comment]
Accepted the changes to get #9519 merged. @makambalaji please take a look on my review comments above.

/lgtm

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-helper-page.ts [hunk 2]
```diff
@@ -9,6 +9,17 @@ export const topologyHelper = {
...
+    cy.get('body').then(($body) => {
+      if (
+        $body.find('[data-test-id="topology-switcher-view"][aria-label="Graph view"]').length !== 0
+      ) {
+        cy.get(topologyPO.list.switcher).click();
+        cy.log('user is switching to graph view in topology page');
+      } else {
+        cy.log('You are on Topology page - Graph view');
+      }
+    });
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-helper-page.ts:22]: In general, the suggested changes makes sense, but cypress doesn't work in that way. Please have a look at below url for more details
https://docs.cypress.io/guides/core-concepts/conditional-testing

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts [hunk 3]
```diff
@@ -24,7 +24,8 @@ export const topologyPage = {
   verifyTopologyPage: () => {
     app.waitForDocumentLoad();
     cy.url().should('include', 'topology');
-    cy.get(topologyPO.graph.emptyGraph).should('be.visible');
+    //  Commenting below line, as above line  already asserts the same
+    // cy.get(topologyPO.graph.emptyGraph).should('be.visible');
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts:28]: Sure, I will remove it in another pr

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-helper-page.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-helper-page.ts:22]: Okay thanks for that link

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts:28]: Thanks

File: frontend/packages/dev-console/integration-tests/.gherkin-lintrc [hunk 4]
```diff
@@ -12,7 +12,7 @@
   "no-empty-file": "on",
   "no-scenario-outlines-without-examples": "on",
   "name-length": "off",
-  "allowed-tags": ["on", {"tags": ["@un-verified", "@knative-admin", "@service-mesh", "@cnv", "@knative-camelk", "@getting-started", "@health-checks", "@e2e", "@to-do", "@manual", "@regression", "@smoke", "@add-flow", "@topology", "@knative", "@guided-tour", "@gitops", "@helm", "@monitoring", "@pipelines", "@virtualization", "@web-terminal", "@vulnerability", "@knative-eventing", "@knative-camelK", "@kafka"]}],
+  "allowed-tags": ["on", {"tags": ["@pre-condition", "@un-verified", "@bug" "@knative-admin", "@service-mesh", "@cnv", "@knative-camelk", "@getting-started", "@health-checks", "@e2e", "@to-do", "@manual", "@regression", "@smoke", "@add-flow", "@topology", "@knative", "@guided-tour", "@gitops", "@helm", "@monitoring", "@pipelines", "@virtualization", "@web-terminal", "@vulnerability", "@knative-eventing", "@knative-camelK", "@kafka"]}],
```
[the remark this answers is not in the document]
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/.gherkin-lintrc:15]: I added due to one of the test case got broken, but bug got fixed, so I removed the tag within this pr
updating "@bug" to "@broken-test" as part of pr:

**Pull request merged**